src/map/entities/battleentity.cpp

```diff
@@ -226,17 +226,44 @@ int32 CBattleEntity::GetMaxMP() const
 
 uint8 CBattleEntity::GetSpeed()
 {
+    uint8 baseSpeed   = speed;
+    uint8 outputSpeed = 0;
+
+    // Mount speed. Independent from regular speed and unaffected by most things.
     // Note: retail treats mounted speed as double what it actually is! 40 is in fact retail accurate!
-    int16 startingSpeed = isMounted() ? 40 + settings::get<int8>("map.MOUNT_SPEED_MOD") : speed;
-    // Mod::MOVE (169)
-    // Mod::MOUNT_MOVE (972)
-    Mod mod = isMounted() ? Mod::MOUNT_MOVE : Mod::MOVE;
+    if (isMounted())
+    {
+        baseSpeed   = 40 + settings::get<int8>("map.MOUNT_SPEED_MOD");
+        outputSpeed = baseSpeed * (100 + getMod(Mod::MOUNT_MOVE)) / 100;
+
+        return std::clamp<uint8>(outputSpeed, 0, 80);
+    }
+
+    // Flee, KIs, Gear penalties, Bolters Roll.
```

Comment:
so bolter's roll fits interestingly into this:

> Stacks multiplicatively with [Movement Speed](https://www.bg-wiki.com/ffxi/Movement_Speed) equipment and additively with either [Mazurka](https://www.bg-wiki.com/ffxi/Mazurka) or [Quickening](https://www.bg-wiki.com/ffxi/Quickening) (but not both).

As usual I've done no retail testing, just wanted to add my 2cents that according to this wording it appears that bolter's roll does need to just be part of the additiveMods here. I.E. the logic LGTM

Reply:
Yeah, I was happy to see that it "fitted nicely" so to say.